dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/ToolMetadata.java

```diff
@@ -50,6 +54,18 @@ private static Map<String, ArgumentMetadata> getArgumentMap(AnnotatedMethod<?> m
                 }
             }
         }
-        return result;
+        return result.isEmpty() ? Collections.emptyMap() : Map.copyOf(result);
```

Comment:
I don't think there's a need to copy here.

Reply:
Ah, ok, I hadn't seen that done this way, and I guess it does make sense to have that map be immutable.